src/main/cpp/src/utils/vid_mapper_sql.cc

```diff
@@ -0,0 +1,178 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "vid_mapper_sql.h"
+#include <ctype.h>
+#include <cstdlib>
+#include <cstdio>
+
+#define VERIFY_OR_THROW(X) if(!(X)) throw SQLBasedVidMapperException(#X);
+
+SQLBasedVidMapper::SQLBasedVidMapper() : VidMapper() {
+  dbi_initialize_r(NULL, &instance);
+  conn = dbi_conn_new_r("pgsql", instance);
+  VERIFY_OR_THROW(conn != NULL);
+
+  dbi_conn_set_option(conn, "host", "localhost");
+  dbi_conn_set_option(conn, "username", "postgres");
+  dbi_conn_set_option(conn, "password", "postgres");
+  dbi_conn_set_option(conn, "dbname", "gendb");
+  dbi_conn_set_option(conn, "encoding", "UTF-8");
```

Comment:
These should be arguments to the constructor. Other arguments include:
* Workspace (string)
* Array (string)